### PR TITLE
Update core .app.src files as able

### PR DESCRIPTION
**Pull request by jamesaimonetti**
More to come but these were the "easy" wins that didn't result in
dependency cycles

### CODE REVIEW

File: kazoo_auth/src/kazoo_auth.app.src [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {application,kazoo_auth,
-             [{applications,[crypto,kazoo_amqp,kazoo_apps,kazoo_caches,
-                             kazoo_data,kazoo_documents,kazoo_stdlib,
-                             kazoo_web,kernel,lager,public_key,stdlib]},
+             [{applications,[crypto,gen_smtp,kazoo_amqp,kazoo_apps,
```
Comment (lazedo) [inline kazoo_auth/src/kazoo_auth.app.src]: `gen_smtp` looks wrong

Comment (jamesaimonetti) [reply kazoo_auth/src/kazoo_auth.app.src]: @lazedo kz_auth_scope uses `binstr:join` :)

Comment (lazedo) [reply kazoo_auth/src/kazoo_auth.app.src]: can we use an alternative ?

Comment (jamesaimonetti) [reply kazoo_auth/src/kazoo_auth.app.src]: done

**Comment by lazedo** [comment]
@jamesaimonetti circleci failing

**Pull request closed without merging**